packages/content/src/raw/index.ts

```diff
@@ -102,11 +102,20 @@ export class API {
    *  - Validates schema + transforms representation -> typed before read
    */
   async get(root: CID, path: string, opts: SchemaOptions): Promise<any> {
+    let value: any;
+    let cid: CID;
     let timerId: ReturnType<typeof setTimeout>;
 
+    try {
+      cid = (await this.#ipfs.dag.resolve(root, { path })).cid;
```

Comment:
Yes it just errors out with "Error: no link named "[path]" under [root]".

Reply:
This is the cause of some issues we are seeing now. `ipfs.dag.resolve` makes a call to the IPFS network to resolve the path. If the `js-ipfs` node is not connected well, this may take awhile and cause the gateway call below to never be called.

Reply:
One solution is when this call times out, we fetch the root object from the gateway (not including the path) and try to resolve the path again. This should not result in errors like #13